include/swift/AST/Decl.h

```diff
@@ -5208,7 +5211,8 @@ class ParamDecl : public VarDecl {
   TypeRepr *TyRepr = nullptr;
 
   struct StoredDefaultArgument {
-    PointerUnion<Expr *, VarDecl *> DefaultArg;
+    PointerUnion<Expr *, VarDecl *> ParsedExprOrVar;
+    Optional<Expr *> TypeCheckedExpr;
```

Comment:
I'm not sure it makes sense to split these up currently. Type checking an expression mutates it, destroying the original value.

Reply:
@slavapestov Ah right, I guess in that case we should just expose something like `getStructuralDefaultExpr` and `getTypeCheckedDefaultExpr` where the former returns either an un-type-checked or a type-checked expr?